atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasServices.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.backup;
+
+import com.palantir.atlasdb.backup.api.AtlasService;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class AtlasServices {
+    private AtlasServices() {
+        // util
+    }
+
+    public static void throwIfAtlasServicesCollide(Set<AtlasService> atlasServices) {
```

Comment:
If I'm not mistaken, this isn't sufficient: if a given AtlasBackup/RestoreService is supposed to only be scoped to a single timelock, then this validates that any given request only contains one, but doesn't check that we aren't currently in the process of backing up a different one. Or is my understanding incorrect?

Reply:
In theory, I guess we could be a bit stronger here, although the **current** internal implementation of the tasks means we won't hit

Backup side:
In practice - `prepareBackup` will be called exactly once per `AtlasBackupService` instance.
In theory - `prepareBackup` should not be called for namespaces who already have an entry in the `inProgressBackups` map, so we could additionally validate for that. 
`completeBackup` is a no-op for services not in this map, so we're protected automatically. 

Restore side:
`prepareRestore` will disable the namespaces in timelock. If the namespaces are already disabled, then we will fail on the timelock side (see DisabledNamespaces#disableAll).
`completeRestore` will re-enable these namespaces, and will similarly fail if there is a conflict.

So the pragmatic change to make here could be to disallow `prepareBackup` for namespaces which already have an in-progress backup.

Reply:
I think that's slightly different to what I'm asking for. I'm asking if we need the validation check here to also check that the timelock service ID on the incoming request also matches everything already held in the various `Map<AtlasService, X>`s that we hold internally.